File: pset2/vigenere/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc,string argv[])
{
    if(argc!=2)
    {
        printf("Usage: ./vigenere k\n");
        exit(1);
    }

    string k = argv[1];
    for(int i=0,n=strlen(k);i<n;i++)
    {
        if(isalpha(k[i]))
        {
            continue;
        }
        else
        {
            printf("Usage: ./vigenere k\n");
            exit(1);
        }
    }
    int l= strlen(k);
    string p = get_string("plaintext: ");

    printf("ciphertext: ");
    int j=0;
    for(int i=0,n=strlen(p);i<n;i++)
    {
        int key;



        if(isalpha(p[i]))
        {

            if(isalpha(k[j%l]))
            {
                if(isupper(k[j%l]))
                {
                    key=k[j%l]-'A';
                }
                if(islower(k[j%l]))
                {
                    key=k[j%l]-'a';
                }
            }

            else
            {
            printf("%c",p[i]);
            continue;
            }

            if(isupper(p[i]))
            {

               printf("%c",(((p[i]-'A')+key)%26)+'A');
               j++;
            }

            if(islower(p[i]))
            {

                printf("%c",(((p[i]-'a')+key)%26)+'a');
                j++;
            }
        }


        else
        {
            printf("%c",p[i]);
        }

    }
    printf("\n");


}

File: pset6/vigenere/vigenere.py
import sys
from cs50 import get_string

if(len(sys.argv)!=2):
     print("Usage: ./vigenere k")
     exit(1)
k=sys.argv[1]

for i in k:
    if(i.isalpha()):
        continue
    else:
        print("Usage: ./vigenere k")
        exit(1)

l=len(k)
p=get_string("plaintext: ")
print("ciphertext:",end=" ")

j=0
key=0
for i in p:
    if(i.isalpha()):
        if(k[j%l].isalpha()):
            if(k[j%l].isupper()):
                key=chr(ord(k[j%l])-ord('A'))
            if(k[j%l].islower()):
                key=chr(ord(k[j%l])-ord('a'))
        else:
            print(i,end="")

        if(i.isupper()):
            print("{}".format(chr((((ord(i)-ord('A'))+ord(key))%26)+ord('A'))),end="")
            j=j+1
        if(i.islower()):
            print("{}".format(chr((((ord(i)-ord('a'))+ord(key))%26)+ord('a'))),end="")
            j=j+1
    else:
        print("{}".format(i),end="")

print()




